packages/blockchain-service/src/BSAggregator.ts

```diff
@@ -36,16 +36,16 @@ export class BSAggregator<
     return this.#blockchainServices.some(bs => bs.validateEncrypted(keyOrJson))
   }
 
-  getBlockchainNameByAddress(address: string): BSCustomName | undefined {
-    return this.#blockchainServices.find(bs => bs.validateAddress(address))?.blockchainName
+  getBlockchainNameByAddress(address: string): BSCustomName[] {
+    return this.#blockchainServices.filter(bs => bs.validateAddress(address)).map(bs => bs.blockchainName)
```

Comment:
I'm ok with this approach but if you have problems with performance, consider change this implementation to use a simple for loop like this:

```typescript
getBlockchainNameByAddress(address: string): BSCustomName | null {
  for (const bs of this.#blockchainServices) {
    if (bs.validateAddress(address)) {
      return bs.blockchainName;
    }
  }
  return null;
}
```